src/Ocelot.Provider.Kubernetes/EndPointClientV1.cs

```diff
@@ -30,8 +33,28 @@ public async Task<EndpointsV1> GetAsync(string serviceName, string kubeNamespace
 
         var response = await Http.GetAsync(request, cancellationToken);
 
-        return response.IsSuccessStatusCode
-            ? await response.ReadContentAsAsync<EndpointsV1>()
-            : null;
+        if (!response.IsSuccessStatusCode)
+        {
+            if (_logger.IsEnabled(LogLevel.Debug))
+            {
+                StatusV1 errorResponse = await response.ReadContentAsAsync<StatusV1>();
+
+                (string resourceKind, string resourceApiVersion) = KubeObjectV1.GetKubeKind<EndpointsV1>();
+                _logger.LogDebug("Failed to retrieve {ResourceApiVersion}/{ResourceKind} {ResourceName} in namespace {ResourceNamespace} ({HttpStatusCode}/{Status}/{StatusReason}): {StatusMessage}",
```

Comment:
Since you log error responses, the more appropriate level is `LoggingLevel.Error`.
```suggestion
                _logger.LogError("Failed to retrieve {ResourceApiVersion}/{ResourceKind} {ResourceName} in namespace {ResourceNamespace} ({HttpStatusCode}/{Status}/{StatusReason}): {StatusMessage}",
```
Examples: [LogError](https://github.com/search?q=repo%3AThreeMammals%2FOcelot%20LogError&type=code)
We can also discuss the use of `LogWarning` due to `IsSuccessStatusCode` being `false`. However, this encompasses a wide range of status codes aka 4xx and 5xx.

Reply:
I think that’s called “message template” format; those expressions are similar to `{0}`, `{1}`, etc, but the names inside the braces become properties of the logged message:

https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-message-template

Reply:
I hadn’t seen the delegate syntax before; I am guessing it’s shorthand for the log-level check I’m doing outside that statement?

Reply:
Oh, I see - this is `IOcelotLogger`, which expects the log message to already have been formatted (and therefore does not take format arguments).

No problem, I can switch to using that instead `ILogger`, if you prefer.

Reply:
Switched to `IOcelotLogger.LogError(Func<string>)`.